### PR TITLE
[HOT-56] Abstract timer logic from FoodTimerList component

### PR DESCRIPTION
**Pull request by Keerush**
JIRA: https://hotpotbuddy.atlassian.net/browse/HOT-56

This PR moves the timer logic out of FoodTimerList and cleans up the code according to those changes:
* FoodTimer folder was combined with FoodTimerList since that component will only used there.
* Created PotView component to hold the data of potContent and cookedPotContent (Will be completed in optimization task to avoid additional re-renders due to timer)
* Intial start on Toggle component to display simple and details view with options of additional views in the future.

### CODE REVIEW

File: client/src/pages/pot-instance/PotInstance.tsx [hunk 1]
```diff
@@ -1,29 +1,59 @@
...
-    const tempArray = cookedPotContent;
-    tempArray.push(food);
-    useCookedPotContent([...tempArray]);
+  const endHotPotSession = () => {
+    setHotpotStart(false);
+  };
+
+  // TODO: should we add time when item was completed to this?
+  const addToCookedPot = (foods: PotContent[]) => {
+    setCookedPotContent([...cookedPotContent, ...foods]);
```
Comment (AndrewLin95) [inline client/src/pages/pot-instance/PotInstance.tsx:33]: How does this work? o_o

File: client/src/pages/pot-instance/components/PotView/PotView.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    addToCookedPot(cookedItems);
+    setPotContent(updatedPotContent);
+  };
+
+  // starts hotpot timer when the "start" button is selected.
+  // Can update to when an ingredient gets added in the future.
+  useEffect(() => {
+    // Pot timers will only start up if there are in progress items in the pot
+    if (potContent.length === 0) {
+      return () => {};
```
Comment (AndrewLin95) [inline client/src/pages/pot-instance/components/PotView/PotView.tsx:48]: What is the point of () => {}?

File: client/src/pages/pot-instance/components/PotView/PotView.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  switch (state) {
+    case PotViewState.Simple:
+      return <div>Simple View</div>;
+    case PotViewState.Detailed:
+      return (
+        <FoodTimerList potContent={potContent} setPotContent={setPotContent} />
+      );
+    default:
+      return <div>Unknown View Selected</div>;
+  }
```
Comment (AndrewLin95) [inline client/src/pages/pot-instance/components/PotView/PotView.tsx:65]: This is cool

**Review by AndrewLin95 — APPROVED** [review]
⚓

File: client/src/pages/pot-instance/PotInstance.tsx (hunk 1 above)
Comment (Keerush) [reply client/src/pages/pot-instance/PotInstance.tsx:33]: The spread operator just pulls out all of the values in the array. 
So spreading multiple arrays together allows you to form a new array that combines values.

File: client/src/pages/pot-instance/components/PotView/PotView.tsx (hunk 2 above)
Comment (Keerush) [reply client/src/pages/pot-instance/components/PotView/PotView.tsx:48]: Addressing eslint error to return a empty state that is in the same form as the return statement below.

**Pull request merged**